### ISSUE TITLE
Easier control over histogram binning

### ISSUE THREAD

**Issue by mjskay** [mjskay/ggdist#118]
Pulling this out as its own issue (original comment from @steveharoz below)

It would be nice to be able to do things like specify things like `center`, `binwidth`, etc (similar to things that `geom_histogram()` supports) instead of `breaks`. Whatever is chosen here should match up with parameters used for #101. Probably this will happen after #113, since that will introduce a mechanism for custom density estimators that would include histograms and allow an expanded set of parameters for those things.

---

Here's a related quirk and feature request:
It'd be great to be able to specify the bin width (not the breaks) for all of the histograms, so it's visually consistent. It would somewhat solve the above problem too.

```
set.seed(1234)
tribble(
  ~group, ~value,
  "b",    rnorm(1000, mean = 6, sd = 3),
  "c",    rnorm(1000, mean = 8, sd = 2),
  "d",    rnorm(1000, mean = 10),
) %>%
  unnest(value)  %>%
  ggplot() +
  aes(y = group, x = value) +
  stat_histinterval(slab_type = "histogram") +
  theme_ggdist()
```

![image](https://user-images.githubusercontent.com/2257540/98964433-14460a80-2509-11eb-951d-efd362424ed0.png)
 
It'd be nice if there were a parameter to specify fixed bin sizes for all histograms. Like `bindwidth = 0.5, binstart=-0.25`. And then all bins would be [ -0.25 + 0.5n, -0.25 + 0.5(n+1) )

_Originally posted by @steveharoz in https://github.com/mjskay/ggdist/issues/32#issuecomment-726175101_

**Comment by mjskay** [comment]
thinking about this more, I'm not a big fan of adding a bunch more options that all interact with each other (ggplot2::stat_bin has bins, center, boundary, and breaks...)

maybe instead functions for different break algorithms that can be passed to `breaks` would do the trick?

**Comment by steveharoz** [comment]
Yeah, I think the key is to make the documentation really obvious and well connected from the histinterval documentation. A consistent naming scheme or prefix across the different breaks functions can help with discoverability.

**Comment by mjskay** [comment]
Okay, I think I've come up with tentative scheme that decouples two things:

1. The algorithm for picking the number of bins / the bin width. This is determined by the `breaks` parameter, which can take the number of bins (a single number), a vector of breakpoints, a function that generates either of those, or the suffix of such a function (as a string --- e.g. `"Sturges"` to specify `breaks_Sturges()`). Several automatic algorithms are provided (`breaks_Scott()`, `breaks_Sturges()`, and `breaks_FD()`). The new `breaks_fixed()` also makes it easy to pick a specific bin width, using `breaks_fixed(width = ...)`.
2. How bins are aligned. This is determined by the `align` parameter, which can take a single number giving an offset to apply to the breaks, a function returning such an offset given the breaks, or the suffix of such a function (as a string --- e.g. `"center"` to specify `align_center()`). Then `align = align_center(at = 0)` would center bins on 0 and `align = align_boundary(at = 0)` would align a bin edge to 0.

An example of control of breaks:

```r
set.seed(1234)
 x = rnorm(200, 1, 2)

 ggplot(data.frame(x), aes(x)) +
   stat_slab(
     aes(y = "fixed at 0.5"),
     density = "histogram",
     breaks = breaks_fixed(width = 0.5),
     outline_bars = TRUE,
     color = "black",
   ) +
   stat_slab(
     aes(y = "Sturges"),
     density = "histogram",
     breaks = "Sturges",
     outline_bars = TRUE,
     color = "black",
   ) +
   stat_slab(
     aes(y = "Scott"),
     density = "histogram",
     breaks = "Scott",
     outline_bars = TRUE,
     color = "black",
   ) +
   stat_slab(
     aes(y = "FD"),
     density = "histogram",
     breaks = "FD",
     outline_bars = TRUE,
     color = "black",
   ) +
   geom_point(aes(y = 0.7), alpha = 0.5)
```
![image](https://user-images.githubusercontent.com/6345019/233873164-7a7bc4d2-96b5-4a6e-bb39-053b8e400d16.png)


And control of alignment:

```r
set.seed(1234)
x = rnorm(200, 1, 2)

ggplot(data.frame(x), aes(x)) +
  stat_slab(
    aes(y = "none"),
    density = "histogram",
    breaks = breaks_fixed(width = 1),
    outline_bars = TRUE,
    # no need to specify align; align_none() is the default
    color = "black",
  ) +
  stat_slab(
    aes(y = "center at 0"),
    density = "histogram",
    breaks = breaks_fixed(width = 1),
    align = align_center(at = 0),   # or align = "center" since 0 is default
    outline_bars = TRUE,
    color = "black",
  ) +
  stat_slab(
    aes(y = "boundary at 0"),
    density = "histogram",
    breaks = breaks_fixed(width = 1),
    align = align_boundary(at = 0), # or align = "boundary" since 0 is default
    outline_bars = TRUE,
    color = "black",
  ) +
  geom_point(aes(y = 0.7), alpha = 0.5)
```
![image](https://user-images.githubusercontent.com/6345019/233873335-a33160a2-66c5-4b94-b8be-4ed8f4a7aaf8.png)

All these can be tested on the (very experimental) dev version installable via `remotes::install_github("mjskay/ggdist@dev")`. Feedback is much appreciated if you have time! Feedback on docs is also appreciated, e.g. the docs for the `align` and `breaks` parameters in `?stat_histinterval`.

**Comment by mjskay** [comment]
I think this is working well on dev now, so closing this.
